diff --git a/Dockerfile b/Dockerfile
index fa56986..4e25db8 100644
--- a/Dockerfile
+++ b/Dockerfile
@@ -5 +5 @@ ARG BUILD_JOBS=16
-ARG CUDA_IMAGE=nvidia/cuda:13.0.2-devel-ubuntu24.04
+ARG CUDA_IMAGE=nvidia/cuda:13.0.2-devel-ubuntu24.04@sha256:5dc1bca23d05bd37b011be68ec470c03b403a5da07ec3a86e41af9470e9d0cc6
@@ -10,0 +11 @@ FROM ${CUDA_IMAGE} AS base
+ENV UV_EXCLUDE_NEWER=2026-06-18T23:59:59Z
@@ -81 +82,2 @@ WORKDIR $VLLM_BASE_DIR
-RUN git clone -b v2.30u1 https://github.com/NVIDIA/nccl.git && \
+RUN git clone https://github.com/NVIDIA/nccl.git && \
+    cd nccl && git checkout 6da422082f910a8dd230f7e42e26ece4dc37bccc && cd .. && \
@@ -346,0 +349 @@ FROM ${CUDA_IMAGE} AS runner
+ENV UV_EXCLUDE_NEWER=2026-06-18T23:59:59Z
@@ -407 +410 @@ RUN --mount=type=bind,source=wheels,target=/workspace/wheels \
-        echo "transformers>=5.0.0" >> /tmp/wheel-override.txt; \
+        echo "transformers==5.12.1" >> /tmp/wheel-override.txt; \
